core/trino-main/src/main/java/io/trino/server/testing/TestingTrinoServer.java

```diff
@@ -439,6 +441,10 @@ public CatalogHandle createCatalog(String catalogName, String connectorName, Map
     {
         CatalogHandle catalogHandle = connectorManager.createCatalog(catalogName, connectorName, properties);
         updateConnectorIdAnnouncement(announcer, catalogHandle, nodeManager);
+
+        EventListenerManager eventListenerManager = injector.getInstance(EventListenerManager.class);
+        connectorManager.getConnectorServices(catalogHandle).getEventListeners().forEach(eventListenerManager::addEventListener);
```

Comment:
Do we need product tests to make sure that event listeners are truly registered?

Reply:
Server correctly registered EventListeners, TestingTrinoServer did not, so it only affected (some) tests